rasa/shared/core/events.py

```diff
@@ -1211,15 +1221,22 @@ def __init__(
 
         super().__init__(timestamp, metadata)
 
-    def __str__(self) -> Text:
+    def __repr__(self) -> Text:
+        """Returns event as string for debugging."""
         return "ActionExecuted(action: {}, policy: {}, confidence: {})".format(
             self.action_name, self.policy, self.confidence
         )
 
+    def __str__(self) -> Text:
+        """Returns event as human readable string."""
+        return self.action_name or self.action_text
```

Comment:
background: [str vs repo](https://stackoverflow.com/a/2626364/3429596)

My idea is that we can use `str` to get a human readable output for things like the story validator and that we limit `as_story_string` to it's usage for dumping markdown.

Reply:
Sounds good to me.

Also, on a separate topic, taking a quote from that article:

> This means, in simple terms: almost every object you implement should have a functional \_\_repr\_\_ that’s usable for understanding the object.

I think this ^ should be a python code convention.